doc/glossary.rst

```diff
@@ -1336,7 +1341,12 @@ Methods
         often the likelihood of the data under the model.
 
     ``score_samples``
-        TODO
+        A method on an array of data points, which evaluates its predictions on
+        the given dataset, and returns an array consisting of log evaluations
```

Comment:
it is not log evaluations for outlier detection and novelty detection estimators. I think a good way to describe score_samples is by saying that it represents the likelihood of the samples:
* for density estimation it's the value of the (log) density of the samples (and hence correspond to the usual definition of likelihood)
* for outlier detection it's a score corresponding to how likely a sample is (an outlier if it's not likely at all): e.g. for the OneClass SVM score_samples is asymptotically proportional to the underlying density in low density regions.